pkg/tsdb/cloudmonitoring/cloudmonitoring.go

```diff
@@ -548,14 +601,25 @@ func (e *Executor) createRequest(ctx context.Context, dsInfo *models.DataSource,
 		}
 	}
 
-	pluginproxy.ApplyRoute(ctx, req, proxyPass, cloudMonitoringRoute, dsInfo, e.cfg)
+	user := &models.SignedInUser{
+		OrgId:   pluginCtx.OrgID,
+		Login:   pluginCtx.User.Login,
+		Name:    pluginCtx.User.Name,
+		Email:   pluginCtx.User.Email,
+		OrgRole: models.RoleType(pluginCtx.User.Role),
+	}
+	ds, err := s.dataSourceCache.GetDatasource(dsInfo.id, user, false)
+	if err != nil {
+		return nil, err
+	}
+
+	pluginproxy.ApplyRoute(ctx, req, proxyPass, cloudMonitoringRoute, ds, s.cfg)
```

Comment:
We discussed this. Had a quick look. Think there are a couple different alternatives. In general JSON data and secure JSON data is what's being used from `models.DataSource`. Seems like perhaps data source id and version is also used and potentially a few other fields. 

Seems like `pluginproxy.ApplyRoute` is only used from two places including this. 

I'm tempted to suggest requiring `pluginproxy.ApplyRoute` to accept a new struct instead of `models.DataSource` which only contains the fields actually being used and let each consumer create this struct. 

The other alternative would be for cloudmonitoring to convert plugin context to `models.DataSource`, but that doesn't feel great and could easily introduce bugs.

WDYT?

Reply:
Yeah I like the former approach. Shouldn't be too difficult!

Reply:
One thing we're missing for this to work @marefr is the data source `version` field on the SDKs `DataSourceInstanceSettings` struct. We'll need to add that in order to address this properly I believe

Reply:
Seems like that is only used for generating access token cache keys. Let me know if it's used for other things. I think we should be able to use the `Updated` field instead from `DataSourceInstanceSettings` and `models.DataSource`. Maybe `Updated.Unix()` in the cache key. WDYT?